borsh/src/de/mod.rs

```diff
@@ -2,24 +2,29 @@ use core::marker::PhantomData;
 use core::mem::MaybeUninit;
 use core::{
     convert::{TryFrom, TryInto},
-    hash::{BuildHasher, Hash},
     mem::size_of,
 };
 
+#[cfg(any(feature = "hashbrown", feature = "std"))]
```

Comment:
Thinking more about "hashbrown" or "std", I feel it might be easier to introduce a single feature that would enable "hash-collections" support, here is what I mean:

Cargo.toml features:

```toml
PRIVATE-hash-collections = []
hash-collections = ["PRIVATE-hash-collections", "hashbrown"]
std = ["PRIVATE-hash-collections"]
```

and then in the code:

```rust
#[cfg(feature = "PRIVATE-hash-collections")]
```

This way "std" won't pull in "hashbrown" dependency. The naming of "PRIVATE-hash-collections" is terrible, so maybe you can come up with a better naming, but at least you don't need to construct this `cfg` rule over and over again. Maybe it is better to use [`cfg_aliases`](https://docs.rs/cfg_aliases/latest/cfg_aliases/)

Reply:
The most meaningful name for `PRIVATE-hash-collections` is `disjunction-vessel-hash-collections`, so i'll try [cfg_aliases](https://docs.rs/cfg_aliases/latest/cfg_aliases) and we'll see how that looks.